### PR TITLE
Add support for omitting entire modules

### PR DESCRIPTION
**Pull request by br3ndonland**
## Description

This PR will enable entire modules to be conditionally omitted from test coverage measurement (#2).

## Changes

Modules will be omitted using the [coverage.py `[run] omit` configuration setting](https://coverage.readthedocs.io/en/stable/config.html#run-omit).

The syntax for omitting modules will be slightly different from the suggestion by @sobolevn (https://github.com/wemake-services/coverage-conditional-plugin/issues/2#issuecomment-1263918296) to align more closely with the file pattern syntax used by coverage.py.

```toml
[tool.coverage.run]
plugins = ["coverage_conditional_plugin"]

[tool.coverage.coverage_conditional_plugin.omit]
"my_project/omit*.py" = "sys_platform == 'win32'"
```

## Related

- Closes wemake-services/coverage-conditional-plugin#2
- https://github.com/encode/uvicorn/issues/1834#issuecomment-1509987813
- [coverage.py docs: configuration reference - `[run] omit`](https://coverage.readthedocs.io/en/stable/config.html#run-omit)
- [coverage.py docs: specifying source files](https://coverage.readthedocs.io/en/stable/source.html)

### CODE REVIEW

File: coverage_conditional_plugin/__init__.py [hunk 1]
```diff
@@ -38,8 +40,23 @@ def configure(self, config: CoverageConfig) -> None:
...
+        try:  # ini format
+            omits = filter(
+                bool,
+                config.get_option(self._omit_opt_name_plugin).splitlines(),
+            )
+        except AttributeError:  # toml format
+            omits_dict = config.get_option(self._omit_opt_name_plugin)
+            omits = omits_dict.items() if omits_dict else ()
+
+        for omit in omits:
```
Comment (sobolevn) [inline coverage_conditional_plugin/__init__.py:56]:
```suggestion
            omits = config.get_option(self._omit_opt_name_plugin) or {}

        for omit in omits.items():
```

**Review by sobolevn — APPROVED** [review]
Thank you! This seems very useful!

I have just one nit to apply :)

**Comment by br3ndonland** [comment]
Thanks for the prompt review @sobolevn!

I'm happy to consider the nit you suggested, but it makes type annotations more complicated, and the `for omit in omits.items():` syntax won't work for the filter returned from the INI file format.

It looks like we need to update `gitpython` to resolve the vulnerability reported by `poetry run safety check --full-report`. I've pushed that update.

**Comment by Kludex** [comment]
What's missing here?

**Comment by sobolevn** [comment]
I can work on it and finish it tomorrow.

**Comment by Kludex** [comment]
> I can work on it and finish it tomorrow.

I didn't mean to give you work, I really wanted to know what was missing hahaha

...but thanks.

**Review by sobolevn — APPROVED** [review]

**Comment by sobolevn** [comment]
Thank you! Great idea!

**Pull request merged**